x/mongo/driver/operation.go

```diff
@@ -1775,8 +1792,8 @@ func (op Operation) publishFinishedEvent(ctx context.Context, info finishedInfor
 	if success {
 		res := bson.Raw{}
 		// Only copy the reply for commands that are not security sensitive
-		if !info.redacted {
-			res = bson.Raw(info.response)
+		if !info.redacted && info.response != nil {
+			res = replyCallback(ctx, info.response)
```

Comment:
I was unable to show a reduction in memory allocations when benchmarking and profiling this change. I believe that's because the copy that was previously made here was removed in https://github.com/mongodb/mongo-go-driver/pull/1053 in favor of copying the response document [here](https://github.com/mongodb/mongo-go-driver/blob/c05cb8aea4a59fda1fffcc3324707bc3664e7ba0/x/mongo/driver/operation.go#L887). As is, the code `res = bson.Raw(info.response)` does not actually create a copy of the response data, just a new slice header.

Removing response copies actually requires changes to `readWireMessage`/`roundTrip`.